src/explorer.api/Queries/SingleColumnHistogram.cs

```diff
@@ -0,0 +1,88 @@
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("explorer.api.tests")]
+
+namespace Explorer.Queries
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.Json;
+
+    using Aircloak.JsonApi;
+    using Aircloak.JsonApi.ResponseTypes;
+
+    internal class SingleColumnHistogram :
+        IQuerySpec<SingleColumnHistogram.Result>
+    {
+        public SingleColumnHistogram(
+            string tableName,
+            string columnName,
+            IList<decimal> buckets)
+        {
+            TableName = tableName;
+            ColumnName = columnName;
+            Buckets = buckets;
+        }
+
+        public string QueryStatement
+        {
+            get
+            {
+                var bucketsFragment = string.Join(
+                    ",",
+                    from bucket in Buckets select $"bucket({ColumnName} by {bucket})");
+
+                return $@"
+                        select
+                            grouping_id(
+                                {bucketsFragment}
+                            ),
+                            {Buckets.Count} as num_buckets,
+                            {bucketsFragment},
+                            count(*),
+                            count_noise(*)
+                        from {TableName}
+                        group by grouping sets ({Enumerable.Range(3, Buckets.Count)})";
+            }
+        }
+
+        public string TableName { get; }
+
+        public string ColumnName { get; }
+
+        public IList<decimal> Buckets { get; }
+
+        public class Result : IJsonArrayConvertible
+        {
+            public decimal? BucketSize { get; set; }
+
+            public AircloakColumn<decimal> LowerBound { get; set; }
```

Comment:
The compiler issues a warning here regarding the uninitialized non-nullable field.

Reply:
Fixed this.